packages/api/pages/api/languages/[code]/verses/[verseId]/next-unapproved-verse.ts

```diff
@@ -0,0 +1,40 @@
+import { GetNextUnapprovedVerse } from '@translation/api-types';
+import createRoute from '../../../../../../shared/Route';
+import { client } from '../../../../../../shared/db';
+import { authorize } from '../../../../../../shared/access-control/authorize';
+
+type NextUnapprovedVerseResult = [{ nextUnapprovedVerseId: string }];
+
+export default createRoute<{ code: string; verseId: string }>()
+  .get<void, GetNextUnapprovedVerse>({
+    authorize: authorize((req) => ({
+      action: 'translate',
+      subject: 'Language',
+      subjectId: req.query.code,
+    })),
+    async handler(req, res) {
+      const language = await client.language.findUnique({
+        where: {
+          code: req.query.code,
+        },
+      });
+      if (!language) {
+        return res.notFound();
+      }
+
+      const [result] = await client.$queryRaw<NextUnapprovedVerseResult>`
+        WITH language_gloss as 
+          (SELECT * FROM "Gloss" 
+            WHERE "Gloss"."languageId" = ${language.id}::uuid)
+        SELECT "Verse"."id" as "nextUnapprovedVerseId" FROM "Verse"
+          JOIN "Word" ON "Word"."verseId" = "Verse"."id"
+          LEFT OUTER JOIN language_gloss ON language_gloss."wordId" = "Word"."id" 
+        WHERE "Verse"."id" > ${req.query.verseId} AND (language_gloss."state" = 'UNAPPROVED' 
+                OR language_gloss."state" IS NULL)
```

Comment:
Two suggestions for the WHERE clause for performance and readability:
* We can use a not equals 'APPROVED' comparison instead of two separate equals for null and unapproved
* We can move the language id filter into this clause and eliminate the subquery for language glosses

Reply:
I couldn't get the first one would work and I believe it is due to some quirks in how null comparison works (see [here](https://learnsql.com/blog/null-comparison-operators/)). 

For the second one, I ended up moving the language ID filter into the `ON` clause for the "Gloss" table. This is because if the language ID filter was in the where clause, only verses with non-NULL unapproved glosses would be returned, since all verse would be joined with English glosses and leave no null entries.

I hope that makes sense.